src/main/java/com/google/sps/GoogleShoppingQuerier.java

```diff
@@ -14,62 +14,113 @@
+  //       Here, "tbs=vw:l" sets that the items are displayed as list and not as grid.
+  // "safe" defines the level of filtering for adult content;
   private final String GOOGLE_SEARCH_BASE_URL =
-      "https://www.google.com/search?tbm=shop&tbs=vw:l&safe=strict&hl=en&source=h";
-  // Above, "tbm=shop" is the search type; "tbs=vw" removes ads and specifies the viewing style;
-  // "hl=en" is the language parameter and "safe=strict" is the parameter for safety check.
+      "https://www.google.com/search?tbm=shop&tbs=vw:l&safe=active";
 
-  public GoogleShoppingQuerier(String shoppingQuery) {
-    this.shoppingQuery = shoppingQuery;
-  }
 
-  public String query() throws IOException, HttpStatusException {
+  /** 
+   * Scrapes Google Shopping based on the input and returns the results.
+   * @param ShoppingQueryInput object, containing fields for the values of search parameters.
+   */
+  public String query(ShoppingQueryInput shoppingQueryInput) throws IOException, ShoppingQuerierConnectionException {
+    // Get the query to be searched and check for validity.
+    String shoppingQuery = shoppingQueryInput.getShoppingQuery();
+
     if (!isValidShoppingQuery(shoppingQuery)) {
-      throw new IOException("Invalid Shopping query.");
+      throw new IllegalArgumentException("Invalid Shopping query.");
     }
 
-    shoppingQuery = cleanValidShoppingQuery(shoppingQuery);
+    // Clean the valid query input before building {@code searchURL}.
+    shoppingQuery = polishShoppingQuery(shoppingQuery);
 
     String query = "q=" + shoppingQuery;
-    String maxResultsNum = "num=6";
-    String searchURL = GOOGLE_SEARCH_BASE_URL + "&" + query + "&" + maxResultsNum;
+    // "num" parameter defines the maximum number of results to return.
+    String maxResultsNumber = "num=" + String.valueOf(shoppingQueryInput.getMaxResultsNumber());
+      // "hl" parameter defines the language to use for the Google search.
+    String language = "hl=" + shoppingQueryInput.getLanguage();
+    
+    String searchURL = GOOGLE_SEARCH_BASE_URL + "&" + query + "&" + language + "&" + maxResultsNumber;
 
-    Document doc;
+    Response response = null;
+    
     try {
-      // Without proper User-Agent, it will result in a 403 error.
-      doc = Jsoup.connect(searchURL).userAgent("Mozilla/5.0").get();
-    } catch (HttpStatusException e) {
-      throw new HttpStatusException(
-          "Error while getting results from Google search", e.getStatusCode(), e.getUrl());
+      // Get the Connection for fetching content from the {@code searchURL}.
+      response = Jsoup.connect(searchURL)
+          .userAgent("Mozilla/5.0")
+          .execute();
+    } catch (MalformedURLException exception) {
+      // Re-throw the exception via ShoppingQuerierConnectionException custom exception.
+      throw new ShoppingQuerierConnectionException(
+          "The request URL must be a HTTP or HTTPS URL and must not be malformed.", exception);
```

Comment:
I think these messages could be improved. 

These messages sound like error messages for wrong input. But that's not the case here. It's better to just describe the error, e.g. all of them can simply be:

"Failed to fetch from Google Search"

That mean this code could be:

catch (Exp1 | Exp2 | ... e) {
  throw new ShoppingQuerierConnectionException("...", e);
}